find.go

```diff
@@ -29,26 +31,51 @@ type UPNPDevice struct {
 	Desc    *deviceDesc
 }
 
+func parseNetworks(queries []DeviceQuery) []string {
+	networks := []string{}
+	for _, query := range queries {
+		if query.Network != "" && slices.Contains(networks, query.Network) {
+			networks = append(networks, query.Network)
+		}
+	}
+	return networks
+}
+
 // FindHost looks for a host matching the query, returns just the host/ip (no port).
-func FindHost(ctx context.Context, logger logging.Logger, query DeviceQuery) (string, error) {
-	all, err := findAll(ctx, logger, query.Network)
+func FindHost(ctx context.Context, logger logging.Logger, queries []DeviceQuery, rootOnly bool) ([]string, error) {
+
+	networks := parseNetworks(queries)
+	all, err := findAll(ctx, logger, networks, rootOnly)
 	if err != nil {
-		return "", err
+		return []string{}, err
 	}
 
+	hostnames := []string{}
 	for _, a := range all {
-		if a.Matches(query) {
-			u, err := url.Parse(a.Service.Location)
-			if err != nil {
-				// should be impossible
-				logger.Warnf("invalid location %s", a.Service.Location)
-				continue
+		for _, query := range queries {
+			if a.Matches(query) {
+				u, err := url.Parse(a.Service.Location)
+				if err != nil {
+					// should be impossible
+					logger.Warnf("invalid location %s", a.Service.Location)
+					continue
+				}
+				if len(query.Endpoints) > 0 {
+					for _, endpoint := range query.Endpoints {
+						hostnames = append(hostnames, fmt.Sprintf("%s/%s", u.Hostname(), endpoint))
+					}
+					continue
+				}
+
+				hostnames = append(hostnames, u.Hostname())
 			}
-			return u.Hostname(), nil
```

Comment:
we no longer return on the first match. Can add this back in some form when only one query exists if requested.